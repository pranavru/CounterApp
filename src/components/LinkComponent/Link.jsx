/* eslint-disable no-alert */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
// Link.react.js
import React from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

const Link = (props) => {
  const [classVal, setClassVal] = React.useState(STATUS.NORMAL);
  const [countValue, setCounter] = React.useState(0);
  function onMouseEnter() {
    setClassVal(STATUS.HOVERED);
  }

  function onMouseLeave() {
    setClassVal(STATUS.NORMAL);
  }

  return (
    <>
      <a
        className={classVal}
        href={props.page || '#'}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      >
        {props.children}
      </a>
      <button type="button" onClick={() => setCounter(countValue + 1)}>
        Counter
        <span>
          {countValue}
        </span>
      </button>
    </>
  );
};

export default Link;
