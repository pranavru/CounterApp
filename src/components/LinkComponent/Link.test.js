/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/jsx-filename-extension */
// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';

import Link from './Link';

Enzyme.configure({ adapter: new Adapter() });
const { act } = renderer;
describe('Link Component:', () => {
  test('Link changes the class when hovered', () => {
    const component = renderer.create(
      <Link page="http://www.facebook.com">Facebook</Link>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    act(() => tree.props.onMouseEnter());
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    act(() => tree.props.onMouseLeave());
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  jest.mock('./Link', () => 'Link');

  test('should simulate click', () => {
    const component = shallow(<Link page="http://www.facebook.com" href="">Facebook</Link>);
    component.find('button').simulate('click');
    expect(component.text()).toBe('FacebookCounter1');
  });
});
