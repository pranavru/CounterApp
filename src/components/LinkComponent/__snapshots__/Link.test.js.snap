// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component: Link changes the class when hovered 1`] = `
Array [
  <a
    className="normal"
    href="http://www.facebook.com"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Facebook
  </a>,
  <button
    onClick={[Function]}
    type="button"
  >
    Counter
    <span>
      0
    </span>
  </button>,
]
`;

exports[`Link Component: Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link Component: Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
